pip/commands/configuration.py

```diff
@@ -0,0 +1,229 @@
+                    # the loop.
+                    action = None
+                    break
+                action = k
+
+        if action is None:
+            logger.warn(
+                "ERROR: Need exactly one action (--list, --edit, "
+                "--get, --set, --unset) to perform."
+            )
+            return ERROR
+
+        # Determine which configuration files are to be loaded
+        if options.user_file and options.global_file:
+            logger.warn(
+                "ERROR: Need at-most one configuration file to use - pass "
+                "only one of --global, --user."
+            )
+            return ERROR
+
+        kwargs = {}
+        if options.user_file:
+            kwargs["load_only"] = "user"
+        elif options.global_file:
+            kwargs["load_only"] = "site-wide"
+        elif options.venv_file:
+            kwargs["load_only"] = "venv"
+        elif action in ["set", "unset", "edit"]:
+            logger.warn(
+                "ERROR: Need one configuration file to modify - pass one of "
+                "--global, --user, --venv."
+            )
+            return ERROR
+
+        # Load a new configuration
+        self.configuration = Configuration(
+            isolated=options.isolated_mode, **kwargs
+        )
+        self.configuration.load()
+
+        # Call the handler for the action with the options
+        handlers = {
+            "list": self.list_values,
+            "edit": self.open_in_editor,
+            "get": self.get_name,
+            "set": self.set_name_value,
+            "unset": self.unset_name
+        }
+
+        return handlers[action](options)
+
+    def list_values(self, options):
+        for key, value in sorted(self.configuration.items()):
+            logger.info("%s=%r", key, value)
+        return SUCCESS
+
+    def open_in_editor(self, options):
+        if options.editor is None:
+            logger.warn(
+                "ERROR: --edit requires an editor to be passed, either using "
+                "--editor or by setting it in a configuration file."
+            )
+            return ERROR
+
+        file = self.configuration.get_file()
+
+        try:
+            subprocess.check_call([options.editor, file])
+        except subprocess.CalledProcessError as e:
+            logger.warn(
+                "ERROR: Subprocess exited with exit code %d", e.returncode
+            )
+            return ERROR
+        else:
+            return SUCCESS
+
+    def get_name(self, options):
+        try:
+            value = self.configuration.get_value(options.get_name)
+        except KeyError:
+            logger.warn("ERROR: No key %r in configuration", options.get_name)
+            return ERROR
+
+        logger.info("%s", value)
+        return SUCCESS
+
+    def set_name_value(self, options):
+        key, value = options.set_name_value.split("=", 1)
+        self.configuration.set_value(key, value)  # Error here is propagated.
```

Comment:
Print a nice message instead of propagating the error? Or is something doing that?

Same on line 212.